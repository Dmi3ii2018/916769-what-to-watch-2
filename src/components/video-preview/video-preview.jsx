import React from "react";
import PropTypes from "prop-types";

export class VideoPreview extends React.PureComponent {
  constructor(props) {
    super(props);

    this._videoRef = React.createRef();
  }

  // componentDidMount() {
  //   const video = this._videoRef.current;

  //   if (this.props.isPlaying) {
  //     video.play();
  //   } else {
  //     video.pause();
  //   }
  // }

  render() {
    const {poster, previewSrc} = this.props;
    return <video ref={this._videoRef} autoPlay muted src={`${previewSrc}`} className="player__video" poster={`img/${poster}.jpg`}>Your browser does not support the video tag.</video>;
  }
}

VideoPreview.propTypes = {
  poster: PropTypes.string,
  previewSrc: PropTypes.string.isRequired,
};
