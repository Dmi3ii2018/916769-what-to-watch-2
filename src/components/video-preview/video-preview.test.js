import React from "react";
import {shallow} from "enzyme";
import {VideoPreview} from "./video-preview";

it(`video player renders properly`, () => {
  const tree = shallow(<VideoPreview
    poster = {`img/macbeth.jpg`}
    previewSrc = {`https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`}
  />);

  expect(tree).toMatchSnapshot();
});
