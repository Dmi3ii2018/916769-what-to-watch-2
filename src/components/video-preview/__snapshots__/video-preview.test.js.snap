// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`video player renders properly 1`] = `ShallowWrapper {}`;
